Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -1433,20 +1404,59 @@ class PropertyWrapperFilter final : public PropertyWrapper<const FilterOperation
             ;
     }
 
-    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation) const final
+    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation compositeOperation) const final
     {
+        auto& fromFilterOperations = value(from);
+        auto& toFilterOperations = value(to);
+
         // https://drafts.fxtf.org/filter-effects/#interpolation-of-filters
-        auto containsReference = [](auto& filterOperations) {
+
+        auto listContainsReference = [](auto& filterOperations) {
             return filterOperations.operations().findIf([](auto& filterOperation) {
                 return filterOperation->type() == FilterOperation::OperationType::REFERENCE;
             }) != notFound;
         };
-        return !containsReference(from.filter()) && !containsReference(to.filter());
+
+        if (listContainsReference(fromFilterOperations) || listContainsReference(toFilterOperations))
+            return false;
+
+        // If one filter is none and the other is a <filter-value-list> without <url>
+        auto oneListIsEmpty = [&]() {
+            return fromFilterOperations.isEmpty() != toFilterOperations.isEmpty();
+        };
+
+        auto listsMatchUpToIndex = [&](auto maxIndex) {
+            ASSERT(fromFilterOperations.size() >= maxIndex && toFilterOperations.size() >= maxIndex);
+            for (size_t i = 0; i < maxIndex; ++i) {
+                auto* fromOperation = fromFilterOperations.at(i);
+                auto* toOperation = toFilterOperations.at(i);
+                if (!!fromOperation != !!toOperation)
+                    return false;
+                if (fromOperation && toOperation && fromOperation->type() != toOperation->type())
+                    return false;
+            }
+
+            return true;
+        };
+
+        // If both filters have a <filter-value-list> of same length without <url> and for each <filter-function>
+        // for which there is a corresponding item in each list
+        auto listsMatchFully = [&]() {
+            return fromFilterOperations.size() == toFilterOperations.size() && listsMatchUpToIndex(fromFilterOperations.size());
+        };
+
+        // If both filters have a <filter-value-list> of different length without <url> and for each
+        // <filter-function> for which there is a corresponding item in each list
+        auto listsMatchPartially = [&]() {
+            return fromFilterOperations.size() != toFilterOperations.size() && listsMatchUpToIndex(std::min(fromFilterOperations.size(), toFilterOperations.size()));
+        };
+
+        return compositeOperation != CompositeOperation::Replace || oneListIsEmpty() || listsMatchFully() || listsMatchPartially();
```

Comment:
`listsMatchFully() || listsMatchPartially()` is potentially two list traversals, and if the lists are the same length (probably very common) they do the same thing. Maybe optimize for lists of the same length.

Reply:
Yes, I can definitely do that.